File: TaskManagement/TaskManagement/Models/project.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace TaskManagement.Models
{
    public enum ProjectStatus
    {
        Active,
        OnHold,
        Completed,
        Archived
    }

    public enum ProjectType
    {
        Software,
        Business,
        Marketing,
        Operations
    }

    public class Project
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(255)]
        public string Name { get; set; } = null!;

        [Required]
        [MaxLength(500)]
        public string ShortDescription { get; set; } = null!;

        [Required]
        [Column(TypeName = "text")]
        public string LongDescription { get; set; } = null!;

        [Required]
        [MaxLength(10)]
        public string ProjectKey { get; set; } = null!;

        public ProjectStatus Status { get; set; }
        public ProjectType Type { get; set; }

        [Required]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        public string LeadId { get; set; } = null!;
        public string? DefaultAssigneeId { get; set; }

        [NotMapped]
        public int Progress
        {
            get
            {
                if (!TaskItems.Any()) return 0;
                var completedTasks = TaskItems.Count(t => t.Status == TaskItemStatus.Done);
                return (int)((double)completedTasks / TaskItems.Count * 100);
            }
        }

        public virtual ICollection<TaskItem> TaskItems { get; set; } = new List<TaskItem>();
        public virtual ICollection<ProjectMember> TeamMembers { get; set; } = new List<ProjectMember>();
    }

    public class ProjectMember
    {
        [Key]
        public int Id { get; set; }
        public string UserId { get; set; } = null!;
        public int ProjectId { get; set; }
        public virtual Project Project { get; set; } = null!;
        public DateTime JoinedAt { get; set; } = DateTime.UtcNow;
        public bool IsActive { get; set; } = true;
    }
}

File: frontend/simplyorganize/src/types/project.ts
// src/types/project.ts

export type ProjectStatus = 'Active' | 'OnHold' | 'Completed' | 'Archived';
export type ProjectType = 'Software' | 'Business' | 'Marketing' | 'Operations';
export type TaskStatus = 'Backlog' | 'ToDo' | 'InProgress' | 'Review' | 'Testing' | 'Done';
export type TaskPriority = 'Low' | 'Medium' | 'High' | 'Critical';
export type TaskType = 'Task' | 'Bug' | 'Story' | 'Epic';

export interface ProjectMember {
  id: number;
  userId: string;
  userName: string;
  userEmail: string;
  joinedAt: string;
  isActive: boolean;
}

export interface TaskComment {
  id: number;
  content: string;
  authorId: string;
  createdAt: string;
}

export interface TaskAttachment {
  id: number;
  fileName: string;
  contentType: string;
  fileSize: number;
  uploadedAt: string;
}

export interface TaskItem {
  id: number;
  title: string;
  shortDescription: string;
  longDescription: string;
  status: TaskStatus;
  priority: TaskPriority;
  type: TaskType;
  createdAt: string;
  updatedAt?: string;
  dueDate?: string;
  assigneeId?: string;
  assigneeName?: string;
  estimatedHours: number;
  labels: string[];
  projectId: number;
  comments: TaskComment[];
  attachments: TaskAttachment[];
}

export interface ProjectSummary {
  id: number;
  name: string;
  projectKey: string;
  shortDescription: string;
  status: ProjectStatus;
  progress: number;
  leadName: string;
  teamMemberCount: number;
  lastUpdated: string;
}

export interface ProjectDetails {
  id: number;
  name: string;
  shortDescription: string;
  longDescription: string;
  projectKey: string;
  status: ProjectStatus;
  type: ProjectType;
  createdAt: string;
  updatedAt?: string;
  leadId: string;
  defaultAssigneeId?: string;
  progress: number;
  totalTasks: number;
  completedTasks: number;
  teamMembers: ProjectMember[];
  tasks: TaskItem[];
}

export interface CreateProjectDto {
  name: string;
  shortDescription: string;
  longDescription: string;
  projectKey: string;
  type: ProjectType;
  leadId: string;
  defaultAssigneeId?: string;
}

export interface UpdateProjectDto {
  name?: string;
  shortDescription?: string;
  longDescription?: string;
  type?: ProjectType;
  status?: ProjectStatus;
  leadId?: string;
  defaultAssigneeId?: string;
}

export interface TasksSummary {
  totalTasks: number;
  completedTasks: number;
  inProgressTasks: number;
  overdueTasks: number;
  upcomingTasks: number;
}